### PR TITLE
Blog post about Kafka failure strategies

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: _posts/2020-10-20-kafka-failure-strategy.adoc [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+But, first, we need to explain how it works under the hood.
+When using reactive messaging, your application receives https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2.4/concepts.html#messages[Messages].
+Even if your method handles _payloads_, there are Messages under the hood, and it unwraps the payload just before calling your method.
+
+image:/assets/images/posts/kafka-failure-strategy/messages.png[]
+
+A message can be _acked_ or _nacked_. 
+If the message processing completes successfully, the message is acknowledged.
+You can manually trigger the acknowledgment (by calling the `ack()` method) or let the framework handles it automatically.
```
Comment (kenfinnigan) [inline _posts/2020-10-20-kafka-failure-strategy.adoc:25]: s/handles/handle

File: _posts/2020-10-20-kafka-failure-strategy.adoc [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+A message can be _acked_ or _nacked_. 
+If the message processing completes successfully, the message is acknowledged.
+You can manually trigger the acknowledgment (by calling the `ack()` method) or let the framework handles it automatically. 
+In general, it's the outbound connector that acknowledges the message once the outgoing message has been sent to the broker successfully.
+If not configured otherwise, acknowledging a message acknowledges the _source_ message, acknowledging its source, until we reach the root message, most probably created by an inbound connector: 
+
+image:/assets/images/posts/kafka-failure-strategy/ack-chain.png[Acknowledgement chain]
+
+When the inbound connector receives the acknowledgment, it can act upon it and, for example, indicates to the broker that the message processing succeeded.
```
Comment (kenfinnigan) [inline _posts/2020-10-20-kafka-failure-strategy.adoc:31]: s/indicates/indicate

File: _posts/2020-10-20-kafka-failure-strategy.adoc [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+When the inbound connector receives the acknowledgment, it can act upon it and, for example, indicates to the broker that the message processing succeeded. 
+In the context of Kafka, there are various commit strategies.
+We will cover these in a future post.
+
+But as said earlier, failures are inevitable. 
+For example, you may have a misbehaving component throwing exceptions, or the outbound connector cannot send the messages because the remote broker is unavailable.
+In these cases, the message is _nacked_, indicating that the processing failed. 
+Similarly to successful acknowledgment, negative acknowledgment can be triggered manually (using the `nack` method) or handled automatically. 
+For example, if your method throws an exception, the message is nacked. 
+As for _ack_, nacking a message nacks its source, and the nack is propagated until the inbound connector:
```
Comment (kenfinnigan) [inline _posts/2020-10-20-kafka-failure-strategy.adoc:40]: Maybe change "for" to "with", for me that creates the tie in to ack

File: _posts/2020-10-20-kafka-failure-strategy.adoc [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+image:/assets/images/posts/kafka-failure-strategy/dead-letter-topic.png[Dead-letter topic]
+
+To enable this strategy, you need to add the following attribute to your configuration:
+[source, text]
+----
+mp.messaging.incoming.movies.failure-strategy=dead-letter-queue
+----
+
+By default, it writes to the `dead-letter-topic-$topic-name` topic.
```
Comment (kenfinnigan) [inline _posts/2020-10-20-kafka-failure-strategy.adoc:188]: Is it worth mentioning whether this topic is created for them by choosing the DLQ strategy or if they need to create it themselves?

Comment (kenfinnigan) [reply _posts/2020-10-20-kafka-failure-strategy.adoc:188]: Not related to the blog, but should the strategy be named DLQ if we're referring to it as DLT everywhere, or it doesn't matter?

**Review by kenfinnigan — CHANGES_REQUESTED** [review]

File: _posts/2020-10-20-kafka-failure-strategy.adoc (hunk 4 above)
Comment (cescoffier) [reply _posts/2020-10-20-kafka-failure-strategy.adoc:188]: Yeah.. not really consistent. I realized that when writing. The pattern is named DLQ, but in Kafka we only have topics (that you can use as a queue). I need to improve it.

Comment (cescoffier) [reply _posts/2020-10-20-kafka-failure-strategy.adoc:188]: Good point, I've added a note. Also, you may need to configure the replication factor.

**Comment by cescoffier** [comment]
@kenfinnigan All done!

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**